core/station/impl/src/services/user_group.rs

```diff
@@ -92,9 +92,29 @@ impl UserGroupService {
     }
 
     pub async fn create(&self, input: AddUserGroupOperationInput) -> ServiceResult<UserGroup> {
-        let user_group_id = generate_uuid_v4().await;
+        self.create_with_id(input, None).await
+    }
+
+    pub async fn create_with_id(
+        &self,
+        input: AddUserGroupOperationInput,
+        with_user_group_id: Option<UUID>,
+    ) -> ServiceResult<UserGroup> {
+        let user_group_uuid = match with_user_group_id {
+            Some(id) => Uuid::from_bytes(id),
+            None => generate_uuid_v4().await,
```

Comment:
Why is this async although the other UUID generations (e.g., for users) are sync (`let user_id = with_id.unwrap_or_else(|| *Uuid::new_v4().as_bytes());`)?

Reply:
Not sure, this `generate_uuid_v4` was probably async for a good reason at some point, but now it seems to be all sync. I'd opt out of extending the scope of the PR to clean this up.